src/Underworld.ts

```diff
@@ -2059,12 +2084,7 @@ export default class Underworld {
       // setTimeout allows the UI to refresh before locking up the CPU with
       // heavy level generation code
       setTimeout(() => {
-        if (this.lastLevelCreated?.levelIndex == levelIndex) {
-          resolve(this.lastLevelCreated);
-          console.warn('Setup: Shortcircuit generateLevelData; returning already generated level data');
-        } else {
```

Comment:
I think this fixed an old bug where if it was called more than once for some reason it wouldn't regenerate... I don't really remember but this was here to protect against something.

Reply:
I figured it might be there to protect against multiple generateLevel() calls in quick succession from players/headless, but this shouldn't happen with the new refactor and the .isHost() check that you implemented before level gen.

This fix was causing other bugs as well, so if changing it somehow introduces a new issue, let's find a different fix

More info: 4150257d25d3f93f26f27bcab1716e479e1b41f6

Reply:
Unfortunately I've found on multiple occasions that what _should_ be impossible happens quite frequently to other players lol.  Let's make sure we keep in protections against any softlocks even if it shouldn't be possible that they occur.

Reply:
Sorry if I wasn't very clear: This old code was **causing** a softlock, not protecting against one.

You can verify this in master by calling the level gen code twice in a row for the same index, which you can do via clicking "skip to X biome" twice. It will get stuck with the "loading" screen overlay

Reply:
> ...but this shouldn't happen with the new refactor and the .isHost() check that you implemented before level gen.

And if this does somehow happen, we should find a fix to prevent it from happening instead of leaving an edge case failsafe that causes other issues, imo

Reply:
ooh okay, that works